--- tiopenmax-pcmdec-3.21.orig/audio/src/openmax_il/pcm_dec/tests/Makefile	2009-04-17 11:12:04.000000000 -0500
+++ tiopenmax-pcmdec-3.21/audio/src/openmax_il/pcm_dec/tests/Makefile	2009-04-17 11:20:49.000000000 -0500
@@ -43,14 +43,13 @@
 
 endif   
 #end of performance code
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_TEST=PcmDecoderTest
 
 COMPONENT_TARGET=$(OMXTESTDIR)/$(COMPONENT_TEST)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
