Index: tiopenmax-aacenc-3.19/audio/src/openmax_il/aac_enc/inc/Makefile
===================================================================
--- tiopenmax-aacenc-3.19.orig/audio/src/openmax_il/aac_enc/inc/Makefile	2009-02-27 11:20:33.000000000 -0600
+++ tiopenmax-aacenc-3.19/audio/src/openmax_il/aac_enc/inc/Makefile	2009-02-27 11:21:33.000000000 -0600
@@ -1,5 +1,5 @@
 
-HSRC=TIDspOmx.h
+HSRC=
 HTARGET=$(HSRC:%=$(OMXINCLUDEDIR)/%)
 
 include $(OMXROOT)/Master.mk
