Index: tiopenmax-aacenc-3.19/audio/src/openmax_il/aac_enc/src/Makefile
===================================================================
--- tiopenmax-aacenc-3.19.orig/audio/src/openmax_il/aac_enc/src/Makefile	2009-02-27 11:15:05.000000000 -0600
+++ tiopenmax-aacenc-3.19/audio/src/openmax_il/aac_enc/src/Makefile	2009-02-27 11:15:19.000000000 -0600
@@ -36,14 +36,13 @@
 
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_LIB=libOMX.TI.AAC.encode.so
 
 COMPONENT_TARGET=$(OMXLIBDIR)/$(COMPONENT_LIB)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR) $(OMXCORE_LIB)
+$(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_LIB)"
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
