Index: meta/packages/exmap-console/exmap-console.inc
===================================================================
--- meta/packages/exmap-console/exmap-console.inc	(revision 5728)
+++ meta/packages/exmap-console/exmap-console.inc	(working copy)
@@ -2,7 +2,7 @@
 HOMEPAGE = "http://projects.o-hand.com/exmap-console"
 SECTION = "devel"
 LICENSE = "GPL"
-DEPENDS = "virtual/kernel readline"
+DEPENDS = "virtual/kernel readline glib-2.0"
 
 SRC_URI = "http://projects.o-hand.com/sources/exmap-console/exmap-console-${PV}.tgz"
 
