Index: meta/classes/sanity.bbclass
===================================================================
--- meta/classes/sanity.bbclass	(revision 5728)
+++ meta/classes/sanity.bbclass	(working copy)
@@ -97,13 +97,6 @@
 		if not check_app_exists("qemu-arm", e.data):
 			messages = messages + "qemu-native was in ASSUME_PROVIDED but the QEMU binaries (qemu-arm) can't be found in PATH"
 
-	if data.getVar('TARGET_ARCH', e.data, True) == "arm":
-		if os.path.exists("/proc/sys/vm/mmap_min_addr"):
-			f = file("/proc/sys/vm/mmap_min_addr", "r")
-			if (f.read().strip() != "0"):
-				messages = messages + "/proc/sys/vm/mmap_min_addr is not 0. This will cause problems with qemu so please fix the value (as root).\n\nTo fix this in later reboots, set vm.mmap_min_addr = 0 in /etc/sysctl.conf.\n"
-			f.close()
-
 	for util in required_utilities.split():
 		if not check_app_exists( util, e.data ):
 			missing = missing + "%s," % util
@@ -125,6 +118,12 @@
 		messages = messages + 'You do not include OpenEmbeddeds version of conf/bitbake.conf. This means your environment is misconfigured, in particular check BBPATH.\n'
 
 	#
+	# Check that the user doesn't have a ';' character in their $PATH
+	#
+	if os.getenv('PATH').find(';') != -1:
+		messages = messages + "your PATH environment variable contains a ';' character"
+
+	#
 	# Check that TMPDIR hasn't changed location since the last time we were run
 	#
 	tmpdir = data.getVar('TMPDIR', e.data, True)
