Index: tiopenmax-videodisplaytest-3.22/application/src/openmax_il/video_display/src/Makefile
===================================================================
--- tiopenmax-videodisplaytest-3.22.orig/application/src/openmax_il/video_display/src/Makefile	2009-05-26 15:42:50.000000000 -0500
+++ tiopenmax-videodisplaytest-3.22/application/src/openmax_il/video_display/src/Makefile	2009-05-26 15:44:19.000000000 -0500
@@ -24,14 +24,13 @@
     CFLAGS += -DOMX_DEBUG=1
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_TEST=VideoDisplayTest
 
 COMPONENT_TARGET=$(OMXTESTDIR)/$(COMPONENT_TEST)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
