Index: tiopenmax-aacdec-3.19/audio/src/openmax_il/aac_dec/tests/Makefile
===================================================================
--- tiopenmax-aacdec-3.19.orig/audio/src/openmax_il/aac_dec/tests/Makefile	2009-02-27 09:21:49.000000000 -0600
+++ tiopenmax-aacdec-3.19/audio/src/openmax_il/aac_dec/tests/Makefile	2009-02-27 09:22:01.000000000 -0600
@@ -36,7 +36,6 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=AacDecoder_Test
 
@@ -44,7 +43,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
