--- tiopenmax-wbamrenc-3.21.orig/audio/src/openmax_il/wbamr_enc/tests/Makefile	2009-04-17 00:33:19.000000000 -0500
+++ tiopenmax-wbamrenc-3.21/audio/src/openmax_il/wbamr_enc/tests/Makefile	2009-04-17 00:41:10.000000000 -0500
@@ -41,7 +41,6 @@
     CFLAGS += -DDSP_RENDERING_ON
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=WbAmrEncTest_common
 
@@ -49,7 +48,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
