--- tiopenmax-ilbcdec-3.21.orig/audio/src/openmax_il/ilbc_dec/tests/Makefile	2009-04-17 13:30:25.000000000 -0500
+++ tiopenmax-ilbcdec-3.21/audio/src/openmax_il/ilbc_dec/tests/Makefile	2009-04-17 13:32:08.000000000 -0500
@@ -43,7 +43,6 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=iLBCDecoderTest
 
@@ -52,7 +51,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
