--- tiopenmax-rageckodec-3.21.orig/audio/src/openmax_il/ragecko_dec/src/Makefile	2009-04-24 02:33:47.000000000 -0500
+++ tiopenmax-rageckodec-3.21/audio/src/openmax_il/ragecko_dec/src/Makefile	2009-04-24 02:34:26.000000000 -0500
@@ -29,14 +29,13 @@
     CFLAGS += -DOMX_DEBUG=1
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_LIB=libOMX.TI.RAG.decode.so
 
 COMPONENT_TARGET=$(OMXLIBDIR)/$(COMPONENT_LIB)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR) $(OMXCORE_LIB)
+$(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_LIB)"
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
