--- tiopenmax-armaacdec-3.20.orig/audio/src/openmax_il/armaac_dec/src/Makefile	2009-04-17 13:49:38.000000000 -0500
+++ tiopenmax-armaacdec-3.20/audio/src/openmax_il/armaac_dec/src/Makefile	2009-04-17 13:51:22.000000000 -0500
@@ -37,18 +37,14 @@
     CFLAGS += -DRESOURCE_MANAGER_ENABLED
 endif
 
-
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_LIB=libOMX.TI.ARMAAC.decode.so
-COMPONENT_LIB_PREBUILT=libEmHeAACDec.so 
-
-
+COMPONENT_LIB_PREBUILT=libEmAACDec.so 
 
 COMPONENT_TARGET=$(OMXLIBDIR)/$(COMPONENT_LIB)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR) $(OMXCORE_LIB)
+$(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_LIB)"
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 	@echo "Installing $(COMPONENT_LIB_PREBUILT)"
