diff -Naur gst-plugins-good-0.10.7.orig/gst/matroska/matroska-demux.c gst-plugins-good-0.10.7/gst/matroska/matroska-demux.c
--- gst-plugins-good-0.10.7.orig/gst/matroska/matroska-demux.c	2008-02-08 09:20:31.000000000 -0600
+++ gst-plugins-good-0.10.7/gst/matroska/matroska-demux.c	2008-12-08 16:37:55.000000000 -0600
@@ -3846,6 +3846,18 @@
           strlen (GST_MATROSKA_CODEC_ID_AUDIO_AC3))) {
     caps = gst_caps_new_simple ("audio/x-ac3", NULL);
     *codec_name = g_strdup ("AC-3 audio");
+  } else if (!strncmp (codec_id, GST_MATROSKA_CODEC_ID_AUDIO_NBAMR,
+          strlen (GST_MATROSKA_CODEC_ID_AUDIO_NBAMR))) {
+    caps = gst_caps_new_simple ("audio/AMR",
+        "rate", G_TYPE_INT, 8000,
+        "channels", G_TYPE_INT, 1, NULL);
+    *codec_name = g_strdup ("AMR-NB audio");
+  } else if (!strncmp (codec_id, GST_MATROSKA_CODEC_ID_AUDIO_WBAMR,
+          strlen (GST_MATROSKA_CODEC_ID_AUDIO_WBAMR))) {
+    caps = gst_caps_new_simple ("audio/AMR-WB",
+        "rate", G_TYPE_INT, 16000,
+        "channels", G_TYPE_INT, 1, NULL);
+    *codec_name = g_strdup ("AMR-WB audio");
   } else if (!strcmp (codec_id, GST_MATROSKA_CODEC_ID_AUDIO_DTS)) {
     caps = gst_caps_new_simple ("audio/x-dts", NULL);
     *codec_name = g_strdup ("DTS audio");
diff -Naur gst-plugins-good-0.10.7.orig/gst/matroska/matroska-ids.h gst-plugins-good-0.10.7/gst/matroska/matroska-ids.h
--- gst-plugins-good-0.10.7.orig/gst/matroska/matroska-ids.h	2007-11-13 11:01:07.000000000 -0600
+++ gst-plugins-good-0.10.7/gst/matroska/matroska-ids.h	2008-12-08 16:38:49.000000000 -0600
@@ -184,6 +184,8 @@
 #define GST_MATROSKA_CODEC_ID_AUDIO_REAL_SIPR    "A_REAL/SIPR"
 #define GST_MATROSKA_CODEC_ID_AUDIO_REAL_RALF    "A_REAL/RALF"
 #define GST_MATROSKA_CODEC_ID_AUDIO_REAL_ATRC    "A_REAL/ATRC"
+#define GST_MATROSKA_CODEC_ID_AUDIO_NBAMR        "A_AMR"
+#define GST_MATROSKA_CODEC_ID_AUDIO_WBAMR        "A_AWB"
 
 /* TODO: AC3-9/10 (?), Musepack, Quicktime */
 
diff -Naur gst-plugins-good-0.10.7.orig/gst/matroska/matroska-mux.c gst-plugins-good-0.10.7/gst/matroska/matroska-mux.c
--- gst-plugins-good-0.10.7.orig/gst/matroska/matroska-mux.c	2008-01-08 14:03:30.000000000 -0600
+++ gst-plugins-good-0.10.7/gst/matroska/matroska-mux.c	2008-12-08 16:40:16.000000000 -0600
@@ -119,6 +119,12 @@
         "endianness = (int) { BIG_ENDIAN, LITTLE_ENDIAN }, "
         "signed = (boolean) true, "
         COMMON_AUDIO_CAPS ";"
+        "audio/AMR, "
+        "rate = (int) 8000, "
+        "channels = (int) 1 ;"
+        "audio/AMR-WB, "
+        "rate = (int) 16000, "
+        "channels = (int) 1 ;"
         "audio/x-tta, "
         "width = (int) { 8, 16, 24 }, "
         "channels = (int) { 1, 2 }, " "rate = (int) [ 8000, 96000 ]")
@@ -1048,6 +1054,14 @@
     context->codec_id = g_strdup (GST_MATROSKA_CODEC_ID_AUDIO_AC3);
 
     return TRUE;
+  } else if (!strcmp (mimetype, "audio/AMR")) {
+    context->codec_id = g_strdup (GST_MATROSKA_CODEC_ID_AUDIO_NBAMR);
+
+    return TRUE;
+  } else if (!strcmp (mimetype, "audio/AMR-WB")) {
+    context->codec_id = g_strdup (GST_MATROSKA_CODEC_ID_AUDIO_WBAMR);
+
+    return TRUE;
   } else if (!strcmp (mimetype, "audio/x-tta")) {
     gint width;
 
