--- tiopenmax-g722dec-3.21.orig/audio/src/openmax_il/g722_dec/src/Makefile	2009-04-17 01:04:19.000000000 -0500
+++ tiopenmax-g722dec-3.21/audio/src/openmax_il/g722_dec/src/Makefile	2009-04-17 01:05:59.000000000 -0500
@@ -34,14 +34,13 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_LIB=libOMX.TI.G722.decode.so
 
 COMPONENT_TARGET=$(OMXLIBDIR)/$(COMPONENT_LIB)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR) $(OMXCORE_LIB)
+$(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_LIB)"
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
