--- tiopenmax-gsmhrenc-3.21.orig/audio/src/openmax_il/gsmhr_enc/tests/Makefile	2009-04-17 03:19:05.000000000 -0500
+++ tiopenmax-gsmhrenc-3.21/audio/src/openmax_il/gsmhr_enc/tests/Makefile	2009-04-17 03:36:52.000000000 -0500
@@ -40,8 +40,6 @@
     CFLAGS += -DDSP_RENDERING_ON
 endif
 
-
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=GSMHrEncTest_common
 
@@ -49,7 +47,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
