--- tiopenmax-armaacenc-3.21.orig/audio/src/openmax_il/armaac_enc/tests/Makefile	2009-04-17 13:39:52.000000000 -0500
+++ tiopenmax-armaacenc-3.21/audio/src/openmax_il/armaac_enc/tests/Makefile	2009-04-17 13:44:05.000000000 -0500
@@ -38,7 +38,6 @@
 
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=ArmAacEncTest_common
 
@@ -46,7 +45,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)  
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
