Index: tiopenmax-3.19/video/src/openmax_il/rv_decode/test/Makefile
===================================================================
--- tiopenmax-3.19.orig/video/src/openmax_il/rv_decode/test/Makefile	2009-02-23 23:28:08.000000000 -0600
+++ tiopenmax-3.19/video/src/openmax_il/rv_decode/test/Makefile	2009-02-23 23:28:40.000000000 -0600
@@ -45,7 +45,7 @@
 
 $(COMPONENT_TEST): $(OBJ)
 #	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -L$(BRIDGELIBDIR) -ldl -lOMX_Core -lpthread -lOMX_RV89Decoder -lbridge -lOMX_ResourceManagerProxy
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lbridge -lOMX_Core -lRM_Parser
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -L$(BRIDGELIBDIR) -lbridge -lOMX_Core -lpthread -lRM_Parser
 
 $(SRC): $(HSRC)
 
