diff -Naurw linux.orig/audio/src/openmax_il/nbamr_enc/inc/OMX_AmrEnc_Utils.h linux/audio/src/openmax_il/nbamr_enc/inc/OMX_AmrEnc_Utils.h
--- linux.orig/audio/src/openmax_il/nbamr_enc/inc/OMX_AmrEnc_Utils.h	2008-10-17 18:24:28.000000000 -0500
+++ linux/audio/src/openmax_il/nbamr_enc/inc/OMX_AmrEnc_Utils.h	2008-11-12 01:43:16.000000000 -0600
@@ -64,7 +64,6 @@
 #include <OMX_Component.h>
 #include "OMX_TI_Common.h"
 #include <TIDspOmx.h>
-#include <ResourceManagerProxyAPI.h>
 
 #ifdef __PERF_INSTRUMENTATION__
     #include "perf.h"
@@ -74,7 +73,7 @@
 	#include <AudioManagerAPI.h>
 #endif
 
-#ifndef UNDER_CE
+#ifdef RESOURCE_MANAGER_ENABLED
 	#include <ResourceManagerProxyAPI.h>
 #endif
 
@@ -716,7 +715,9 @@
     OMX_PORT_PARAM_TYPE* sPortParam;
     OMX_PRIORITYMGMTTYPE* sPriorityMgmt;
 	
+#ifdef RESOURCE_MANAGER_ENABLED
 	RMPROXY_CALLBACKTYPE rmproxyCallback;
+#endif
     OMX_BOOL bPreempted;
 	
     OMX_PARAM_PORTDEFINITIONTYPE* pPortDef[NBAMRENC_NUM_OF_PORTS];
@@ -1180,7 +1181,9 @@
 						 OMX_DIRTYPE eDir);
 
 						 
+#ifdef RESOURCE_MANAGER_ENABLED
 void NBAMRENC_ResourceManagerCallback(RMPROXY_COMMANDDATATYPE cbData);
+#endif
 
 /* ======================================================================= */
 /** OMX_NBAMRENC_INDEXAUDIOTYPE  Defines the custom configuration settings
diff -Naurw linux.orig/audio/src/openmax_il/nbamr_enc/src/OMX_AmrEnc_Utils.c linux/audio/src/openmax_il/nbamr_enc/src/OMX_AmrEnc_Utils.c
--- linux.orig/audio/src/openmax_il/nbamr_enc/src/OMX_AmrEnc_Utils.c	2008-10-17 19:11:57.000000000 -0500
+++ linux/audio/src/openmax_il/nbamr_enc/src/OMX_AmrEnc_Utils.c	2008-11-12 01:43:38.000000000 -0600
@@ -3183,7 +3183,7 @@
 }
 
 
-
+#ifdef RESOURCE_MANAGER_ENABLED
 void NBAMRENC_ResourceManagerCallback(RMPROXY_COMMANDDATATYPE cbData)
 {
     OMX_COMMANDTYPE Cmd = OMX_CommandStateSet;
@@ -3212,3 +3212,4 @@
     }
 
 }
+#endif
