--- tiopenmax-0.3/video/src/openmax_il/post_processor/inc/OMX_PostProc_Utils.h.orig	2008-08-05 14:07:22.000000000 -0500
+++ tiopenmax-0.3/video/src/openmax_il/post_processor/inc/OMX_PostProc_Utils.h	2008-10-28 19:33:08.000000000 -0500
@@ -41,7 +41,9 @@
 #include <OMX_Component.h>
 #include "OMX_PostProc.h"
 #include "OMX_PostProc_CustomCmd.h"
+#ifdef RESOURCE_MANAGER_ENABLED
 #include <ResourceManagerProxyAPI.h>
+#endif
 
 #define POSTPROC_NUM_CUSTOM_PARAMS 7
 
@@ -436,7 +438,9 @@
 	/* used by priority mgr */
 	OMX_PRIORITYMGMTTYPE* pPriorityMgmt;
 	/* used by RM callback */
+#ifdef RESOURCE_MANAGER_ENABLED
     RMPROXY_CALLBACKTYPE rmproxyCallback;
+#endif
 	/* used to keep track of preempted state */
     OMX_BOOL bPreempted;
 
@@ -472,7 +476,9 @@
 OMX_ERRORTYPE RequestPostProcKernelBuffers(OMX_POSTPROC_COMPONENT_PRIVATE *pComponentPrivate);
 OMX_ERRORTYPE PostProcSetVideoPipelines (OMX_POSTPROC_COMPONENT_PRIVATE *pComponentPrivate);
 OMX_ERRORTYPE PostProcSetOutputDevices(OMX_POSTPROC_COMPONENT_PRIVATE *pComponentPrivate);
+#ifdef RESOURCE_MANAGER_ENABLED
 void PostProc_ResourceManagerCallback(RMPROXY_COMMANDDATATYPE cbData);
+#endif
 
 #endif /* OMX_POSTPROC_UTILS_H */
 
--- tiopenmax-0.3/video/src/openmax_il/post_processor/src/OMX_PostProc_Utils.c.orig	2008-08-11 10:14:52.000000000 -0500
+++ tiopenmax-0.3/video/src/openmax_il/post_processor/src/OMX_PostProc_Utils.c	2008-10-28 19:35:35.000000000 -0500
@@ -4079,6 +4079,7 @@
 }
 /* End of ReturnAllBuffer */
 
+#ifdef RESOURCE_MANAGER_ENABLED
 void PostProc_ResourceManagerCallback(RMPROXY_COMMANDDATATYPE cbData)
 {
     OMX_COMMANDTYPE Cmd = OMX_CommandStateSet;
@@ -4137,3 +4138,4 @@
    	  	write(pCompPrivate->nCmdDataPipe[1], &state, sizeof(OMX_U32)) ;
     }
 }
+#endif
