Index: tiopenmax-3.19/audio/src/openmax_il/Makefile
===================================================================
--- tiopenmax-3.19.orig/audio/src/openmax_il/Makefile	2009-02-03 14:09:19.000000000 -0600
+++ tiopenmax-3.19/audio/src/openmax_il/Makefile	2009-03-03 13:30:07.000000000 -0600
@@ -33,8 +33,7 @@
 	gsmfr_dec\
 	gsmfr_enc\
 	gsmhr_dec\
-	gsmhr_enc\
-	ragecko_dec
+	gsmhr_enc
 	
 $(BASETARGETS)::
 	@$(call traverse_dirs,$(DIRS),$@)
Index: tiopenmax-3.19/video/src/openmax_il/Makefile
===================================================================
--- tiopenmax-3.19.orig/video/src/openmax_il/Makefile	2009-03-03 13:31:40.000000000 -0600
+++ tiopenmax-3.19/video/src/openmax_il/Makefile	2009-03-03 13:31:58.000000000 -0600
@@ -12,8 +12,7 @@
 	video_encode \
 	video_decode \
 	rm_parser \
-	rm_rvparser \
-	rv_decode
+	rm_rvparser
 
 
 EXTRA= \
