Index: tiopenmax-3.20/video/src/openmax_il/Makefile
===================================================================
--- tiopenmax-3.20.orig/video/src/openmax_il/Makefile	2009-03-27 16:03:57.000000000 -0500
+++ tiopenmax-3.20/video/src/openmax_il/Makefile	2009-03-27 16:03:59.000000000 -0500
@@ -10,10 +10,7 @@
 	camera \
 	prepost_processor \
 	video_encode \
-	video_decode \
-	rm_parser \
-	rm_rvparser \
-	rv_decode
+	video_decode
 
 
 EXTRA= \
Index: tiopenmax-3.20/audio/src/openmax_il/Makefile
===================================================================
--- tiopenmax-3.20.orig/audio/src/openmax_il/Makefile	2009-03-27 16:03:57.000000000 -0500
+++ tiopenmax-3.20/audio/src/openmax_il/Makefile	2009-03-27 16:03:59.000000000 -0500
@@ -35,8 +35,7 @@
 	gsmhr_dec\
 	gsmhr_enc\
 	imaadpcm_dec\
-	imaadpcm_enc\
-	ragecko_dec
+	imaadpcm_enc
 	
 $(BASETARGETS)::
 	@$(call traverse_dirs,$(DIRS),$@)
Index: tiopenmax-3.20/application/src/openmax_il/avplay/src/Makefile
===================================================================
--- tiopenmax-3.20.orig/application/src/openmax_il/avplay/src/Makefile	2009-03-27 16:03:57.000000000 -0500
+++ tiopenmax-3.20/application/src/openmax_il/avplay/src/Makefile	2009-03-27 16:03:59.000000000 -0500
@@ -6,8 +6,7 @@
 
 SRC=\
 	AvPlayTest.c \
-	AvPlayTest_WMV.c \
-    AvPlayTest_Real.c
+	AvPlayTest_WMV.c
 
 HSRC=$(wildcard ../inc/*)
 
Index: tiopenmax-3.20/application/src/openmax_il/avplay/src/AvPlayTest.c
===================================================================
--- tiopenmax-3.20.orig/application/src/openmax_il/avplay/src/AvPlayTest.c	2009-03-27 16:03:57.000000000 -0500
+++ tiopenmax-3.20/application/src/openmax_il/avplay/src/AvPlayTest.c	2009-03-27 16:07:38.000000000 -0500
@@ -113,7 +113,6 @@
 int Test_AVSync_Video_BAMR(int argc, char** argv);
 int Test_AVSync_Video_AAC(int argc, char** argv, int bARMCodec);
 extern int WmaWmvTest(int argc, char** argv);
-extern int Test_AVSync_Video_RA(int argc, char** argv);
 
 int nPostProcInPipe[2];
 int VPP_IpBuf_Pipe[2];
@@ -652,10 +651,6 @@
         /* With MP3 */
         eError = Test_AVSync_Video_MPEG1L2(argc,argv);
     }
-    else if (atoi(argv[1]) == 18) {
-        /* With RA */
-        eError = Test_AVSync_Video_RA(argc,argv);
-    }
 
 
 	muntrace();
