diff -Naur tiopenmax-jpegdec-3.17.orig/image/src/openmax_il/jpeg_dec/tests/Makefile tiopenmax-jpegdec-3.17/image/src/openmax_il/jpeg_dec/tests/Makefile
--- tiopenmax-jpegdec-3.17.orig/image/src/openmax_il/jpeg_dec/tests/Makefile	2008-10-22 11:00:15.000000000 -0500
+++ tiopenmax-jpegdec-3.17/image/src/openmax_il/jpeg_dec/tests/Makefile	2008-12-02 00:23:50.000000000 -0600
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lbridge -lOMX_Core 
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lpthread -lbridge -lOMX_Core 
 $(SRC): $(HSRC)
 
 clean:: 
