--- tiopenmax-ilbcenc-3.21.orig/audio/src/openmax_il/ilbc_enc/tests/Makefile	2009-04-17 13:15:02.000000000 -0500
+++ tiopenmax-ilbcenc-3.21/audio/src/openmax_il/ilbc_enc/tests/Makefile	2009-04-17 13:23:53.000000000 -0500
@@ -41,7 +41,6 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=iLBCEncTest_common
 
@@ -49,7 +48,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)  
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
