Index: u-boot-5.06/cpu/omap3/config.mk
===================================================================
--- u-boot-5.06.orig/cpu/omap3/config.mk
+++ u-boot-5.06/cpu/omap3/config.mk
@@ -23,7 +23,7 @@
 PLATFORM_RELFLAGS += -fno-strict-aliasing  -fno-common -ffixed-r8 \
 	-msoft-float
 
-PLATFORM_CPPFLAGS += -march=armv7a
+PLATFORM_CPPFLAGS += -march=armv7-a
 # =========================================================================
 #
 # Supply options according to compiler version
