--- tiopenmax-imaadpcmenc-3.21.orig/audio/src/openmax_il/imaadpcm_enc/tests/ImaAdpcmEncTest.mk	2009-04-24 04:18:52.000000000 -0500
+++ tiopenmax-imaadpcmenc-3.21/audio/src/openmax_il/imaadpcm_enc/tests/ImaAdpcmEncTest.mk	2009-04-24 04:40:30.000000000 -0500
@@ -1,5 +1,5 @@
-TEST_COMPONENT = G726EncTest
-OBJ_FILES = G726EncTest.o 
+TEST_COMPONENT = IMAEncTest
+OBJ_FILES = IMAEncTest.o 
 
 # library files to link with when building
 LIBRARIES += -ldl -lLCML -lOMX_Core -lpthread -lOMX_AmrEncoder
