diff -Naur dspbridge/mpu_api.orig/make/build.mk dspbridge/mpu_api/make/build.mk
--- dspbridge/mpu_api.orig/make/build.mk	2008-06-11 16:40:36.000000000 -0500
+++ dspbridge/mpu_api/make/build.mk	2008-08-14 15:57:01.000000000 -0500
@@ -79,7 +79,7 @@
 #   Machine dependent
 
 ifeq ($(PROCFAMILY),OMAP_3430)
-CFLAGS += -mapcs -mno-sched-prolog -mabi=aapcs-linux -mno-thumb-interwork -march=armv7a -msoft-float -Uarm -D"KBUILD_STR(s)=\#s" -D"KBUILD_BASENAME=KBUILD_STR($(basename $(TARGETNAME)))" -D"KBUILD_MODNAME=KBUILD_STR($(basename $(TARGETNAME)))"  -DMODULE -D__LINUX_ARM_ARCH__=7 
+CFLAGS += -mapcs -mno-sched-prolog -mabi=aapcs-linux -mno-thumb-interwork -march=armv7-a -msoft-float -Uarm -D"KBUILD_STR(s)=\#s" -D"KBUILD_BASENAME=KBUILD_STR($(basename $(TARGETNAME)))" -D"KBUILD_MODNAME=KBUILD_STR($(basename $(TARGETNAME)))"  -DMODULE -D__LINUX_ARM_ARCH__=7 
 endif
 
 #   Code generation
diff -Naur dspbridge/mpu_api.orig/make/start.mk dspbridge/mpu_api/make/start.mk
--- dspbridge/mpu_api.orig/make/start.mk	2008-06-11 16:40:42.000000000 -0500
+++ dspbridge/mpu_api/make/start.mk	2008-08-14 16:05:01.000000000 -0500
@@ -32,7 +32,9 @@
 CMDDEFS_START =
 
 
+ifndef CROSS
 CROSS=arm-none-linux-gnueabi-
+endif
 PROCFAMILY=OMAP_3430
 
 
@@ -91,8 +93,15 @@
 %.h:
 
 #   Tools
+ifndef CC
 CC	:= $(CROSS)gcc
+endif
+ifndef AR
 AR	:= $(CROSS)ar
+endif
+ifndef LD
 LD	:= $(CROSS)ld
+endif
+ifndef STRIP
 STRIP	:= $(CROSS)strip
-
+endif
diff -Naur dspbridge/mpu_api.orig/src/bridge/Makefile dspbridge/mpu_api/src/bridge/Makefile
--- dspbridge/mpu_api.orig/src/bridge/Makefile	2008-05-16 04:58:52.000000000 -0500
+++ dspbridge/mpu_api/src/bridge/Makefile	2008-08-14 16:31:18.000000000 -0500
@@ -90,7 +90,7 @@
 # Libraries needed for linking.
 
 ST_LIBS        =
-SH_LIBS        = pthread
+SH_LIBS        =
 
 # Search path for library (and linker command) files.
 # Current folder and target folder are included by default.
diff -Naur dspbridge/mpu_api.orig/src/qos/Makefile dspbridge/mpu_api/src/qos/Makefile
--- dspbridge/mpu_api.orig/src/qos/Makefile	2008-03-06 07:36:36.000000000 -0600
+++ dspbridge/mpu_api/src/qos/Makefile	2008-08-14 17:57:45.000000000 -0500
@@ -89,7 +89,7 @@
 # Libraries needed for linking.
 
 ST_LIBS        =
-SH_LIBS        = pthread
+SH_LIBS        =
 
 # Search path for library (and linker command) files.
 # Current folder and target folder are included by default.
