diff -Naur dspbridge/mpu_driver.orig/make/build.mk dspbridge/mpu_driver/make/build.mk
--- dspbridge/mpu_driver.orig/make/build.mk	2008-06-11 17:12:40.000000000 -0500
+++ dspbridge/mpu_driver/make/build.mk	2008-08-14 12:48:48.000000000 -0500
@@ -79,7 +79,7 @@
 #   Machine dependent
 
 ifeq ($(PROCFAMILY),OMAP_3430)
-EXTRA_CFLAGS += -mapcs -mno-sched-prolog -mabi=aapcs-linux -mno-thumb-interwork -march=armv7a -msoft-float -Uarm -D"KBUILD_STR(s)=\#s" -D"KBUILD_BASENAME=KBUILD_STR($(basename $(TARGETNAME)))" -D"KBUILD_MODNAME=KBUILD_STR($(basename $(TARGETNAME)))"  -DMODULE -D__LINUX_ARM_ARCH__=7 
+EXTRA_CFLAGS += -mapcs -mno-sched-prolog -mabi=aapcs-linux -mno-thumb-interwork -march=armv7-a -msoft-float -Uarm -D"KBUILD_STR(s)=\#s" -D"KBUILD_BASENAME=KBUILD_STR($(basename $(TARGETNAME)))" -D"KBUILD_MODNAME=KBUILD_STR($(basename $(TARGETNAME)))"  -DMODULE -D__LINUX_ARM_ARCH__=7 
 endif
 
 #   Code generation
@@ -216,7 +216,7 @@
 	$(CC) -c $(TARGETARGS) $(EXTRA_CFLAGS) $(INCPATH) -o $@ $<
 
 # removed - need to be done as a pre-step to building
-	$(MAKE) -C $(KRNLSRC) M=$(PROJROOT)/src modules	
+	LDFLAGS= CROSS_COMPILE=$(CROSS) $(MAKE) -C $(KRNLSRC) M=$(PROJROOT)/src modules	
 
 # Generate Module vermagic
 $(TARGETMOD).c: $(TARGET) 
diff -Naur dspbridge/mpu_driver.orig/make/start.mk dspbridge/mpu_driver/make/start.mk
--- dspbridge/mpu_driver.orig/make/start.mk	2008-06-11 17:12:48.000000000 -0500
+++ dspbridge/mpu_driver/make/start.mk	2008-08-13 19:28:36.000000000 -0500
@@ -65,19 +65,23 @@
 
 BRIDGE_BLD_LINUX_VERSION = $(shell grep LINUX_VERSION_CODE $(KRN_VERSION) | cut -d ' ' -f 3 | cut -c1-4 )
 
-
-ifdef CONFIG_ARCH_OMAP34XX
-PROCFAMILY=OMAP_3430
+ifndef CROSS
 ifeq ($(KRN_VER),2.4)
 CROSS=arm_v5t_le-
 endif
+ifeq ($(KRN_VER),2.6)
+CROSS=arm-none-linux-gnueabi-
+endif
+endif
+
+ifdef CONFIG_ARCH_OMAP34XX
+PROCFAMILY=OMAP_3430
 ifdef CONFIG_PM
 CMDDEFS_START +=
 else
 CMDDEFS_START +=DISABLE_BRIDGE_PM
 endif
 ifeq ($(KRN_VER),2.6)
-CROSS=arm-none-linux-gnueabi-
 #CMDDEFS_START +=RES_CLEANUP_DISABLE
 CMDDEFS_START += OPT_ELIMINATE_EXTRA_DLOAD
 CMDDEFS_START +=OPT_REDUCE_SYMBOL_LOOKUPS
@@ -158,10 +162,18 @@
 %.h:
 
 #   Tools
+ifndef CC
 CC	:= $(CROSS)gcc
+endif
+ifndef AR
 AR	:= $(CROSS)ar
+endif
+ifndef LD
 LD	:= $(CROSS)ld
+endif
+ifndef STRIP
 STRIP	:= $(CROSS)strip
+endif
 ifneq ($(wildcard $(KRNLSRC)/scripts/modpost),)
 MODPOST := $(KRNLSRC)/scripts/modpost		\
         $(if $(CONFIG_MODVERSIONS),-m)             \
diff -Naur dspbridge/mpu_driver.orig/src/Makefile dspbridge/mpu_driver/src/Makefile
--- dspbridge/mpu_driver.orig/src/Makefile	2008-04-18 13:53:44.000000000 -0500
+++ dspbridge/mpu_driver/src/Makefile	2008-08-14 12:37:14.000000000 -0500
@@ -15,7 +15,9 @@
 # General Public License for more details.
 #
 
+ifndef PROJROOT
 PROJROOT	= ..
+endif
 
 include $(PROJROOT)/make/start.mk
 # Do not change above "include" line(s)
