diff -Naur tiopenmax-jpegenc-3.17.orig/image/src/openmax_il/jpeg_enc/src/Makefile tiopenmax-jpegenc-3.17/image/src/openmax_il/jpeg_enc/src/Makefile
--- tiopenmax-jpegenc-3.17.orig/image/src/openmax_il/jpeg_enc/src/Makefile	2008-10-22 11:00:39.000000000 -0500
+++ tiopenmax-jpegenc-3.17/image/src/openmax_il/jpeg_enc/src/Makefile	2008-12-02 00:39:43.000000000 -0600
@@ -25,14 +25,13 @@
     CFLAGS += -DOMX_DEBUG=1
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_LIB=libOMX.TI.JPEG.encoder.so
 
 COMPONENT_TARGET=$(OMXLIBDIR)/$(COMPONENT_LIB)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR) $(OMXCORE_LIB)
+$(COMPONENT_TARGET): $(COMPONENT_LIB) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_LIB)"
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
