diff -Naur tiopenmax-jpegenc-3.17.orig/image/src/openmax_il/jpeg_enc/test/Makefile tiopenmax-jpegenc-3.17/image/src/openmax_il/jpeg_enc/test/Makefile
--- tiopenmax-jpegenc-3.17.orig/image/src/openmax_il/jpeg_enc/test/Makefile	2008-10-22 11:01:03.000000000 -0500
+++ tiopenmax-jpegenc-3.17/image/src/openmax_il/jpeg_enc/test/Makefile	2008-12-02 00:44:03.000000000 -0600
@@ -46,10 +46,10 @@
 	cp -f $(COMPONENT_TEST_MULTIPLE) $(COMPONENT_TARGET_MULTIPLE)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lbridge -lOMX_Core  
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lpthread -lbridge -lOMX_Core
 
 $(COMPONENT_TEST_MULTIPLE): $(OBJ_MUL)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST_MULTIPLE) $(OBJ_MUL) -L$(OMXLIBDIR) -lbridge -lOMX_Core  
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST_MULTIPLE) $(OBJ_MUL) -L$(OMXLIBDIR) -lpthread -lbridge -lOMX_Core
 
 $(SRC): $(HSRC)
 
