Index: gst-goo-5.12-P1/src/gstgoovideofilter.c
===================================================================
--- gst-goo-5.12-P1.orig/src/gstgoovideofilter.c
+++ gst-goo-5.12-P1/src/gstgoovideofilter.c
@@ -983,7 +983,6 @@ gst_goo_video_filter_setcaps (GstPad* pa
 {
 	GstGooVideoFilter* self;
 	GstGooVideoFilterClass *klass;
-	GstGooVideoFilterPrivate* priv;
 	GstPad* otherpad;
 	GstPad* otherpeer;
 	GstCaps* othercaps = NULL;
@@ -996,7 +995,6 @@ gst_goo_video_filter_setcaps (GstPad* pa
 	GST_DEBUG_OBJECT (self, "");
 
 	g_return_val_if_fail (GST_CAPS_IS_SIMPLE (caps), FALSE);
-	g_return_val_if_fail (priv != NULL, FALSE);
 
 	otherpad = (pad == self->srcpad) ? self->sinkpad : self->srcpad;
 	otherpeer = gst_pad_get_peer (otherpad);
