--- tiopenmax-wbamrdec-3.20.orig/audio/src/openmax_il/wbamr_dec/tests/Makefile	2009-04-17 00:49:34.000000000 -0500
+++ tiopenmax-wbamrdec-3.20/audio/src/openmax_il/wbamr_dec/tests/Makefile	2009-04-17 00:55:46.000000000 -0500
@@ -37,7 +37,6 @@
     CFLAGS += -DDSP_RENDERING_ON
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=WBAmrDecTest_common
 
@@ -45,7 +44,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) 
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
