diff -Naur amrnb-7.0.0.2.orig/configure.ac amrnb-7.0.0.2/configure.ac
--- amrnb-7.0.0.2.orig/configure.ac	2008-05-21 17:21:50.000000000 -0500
+++ amrnb-7.0.0.2/configure.ac	2008-12-02 16:53:43.000000000 -0600
@@ -21,7 +21,8 @@
 if test x"$PATCH" = x -a ! -f "sp_enc.c" ; then
     AC_MSG_ERROR([You need patch utility to prepare sources.])
 fi
-AC_PARAMETER_EXPANSION_STRING_REPLACE_CAPABLE_SHELL
+# AC_PARAMETER_EXPANSION_STRING_REPLACE_CAPABLE_SHELL
+AC_PATH_PROG([PARAMETER_EXPANSION_STRING_REPLACE_CAPABLE_SHELL],[bash])
 if test x"$PARAMETER_EXPANSION_STRING_REPLACE_CAPABLE_SHELL" = x -a ! "(" -f "26104-700.zip" -o -f "sp_enc.c" ")" ; then
 dnl number sigh cannot be a part of AC_MSG_ERROR
     NUMBER_SIGN="#"
