--- tiopenmax-g723dec-3.21.orig/audio/src/openmax_il/g723_dec/tests/Makefile	2009-04-17 02:12:29.000000000 -0500
+++ tiopenmax-g723dec-3.21/audio/src/openmax_il/g723_dec/tests/Makefile	2009-04-17 02:24:15.000000000 -0500
@@ -37,7 +37,6 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=G723DecTest
 
@@ -45,7 +44,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)  
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
