--- tiopenmax-g726enc-3.21.orig/audio/src/openmax_il/g726_enc/tests/Makefile	2009-04-17 03:07:57.000000000 -0500
+++ tiopenmax-g726enc-3.21/audio/src/openmax_il/g726_enc/tests/Makefile	2009-04-17 03:30:41.000000000 -0500
@@ -42,7 +42,6 @@
 endif
 
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 OMXRMPROXY_LIB=$(OMXLIBDIR)/libOMX_ResourceManagerProxy.so
 COMPONENT_TEST=G726EncTest
 
@@ -50,7 +49,7 @@
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)  
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
